mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java

```diff
@@ -230,8 +231,9 @@
                                    BaseControlCenterPagerAdapter.ControlCenterCallbacks,
                                    AntiPhishing.AntiPhishingCallback,
                                    AntiPhishingDialog.AntiPhishingDialogActionListener,
+                                   OnboardingHelper.OnboardingListener,
+                                   SharedPreferences.OnSharedPreferenceChangeListener {
```

Comment:
Why do we need a listener when `onResume` get called every time? Can not we avoid to add all these error/leak prone listeners?
More, if you have too add the listener in the block instead of after wrapping also the on boarding listener.